api/test/unit/services/application.service.spec.ts

```diff
@@ -768,6 +773,108 @@ describe('Testing application service', () => {
     });
   });
 
```

Comment:
Love to get some advice on the best approach to unit testing here. Worried this one doesn't cover anything beyond the e2e but equally worried of too much mocking and also not getting much coverage.